Fixed missing index in RatingList. Updated averageReviewRating get ro…

nelsonly123 opened the pull request:
…ute to reduce the number of decimals in ratings from 2 to 1.

nelsonly123 committed:
Fixed missing index in RatingList. Updated averageReviewRating get route to reduce the number of decimals in ratings from 2 to 1.

GalexyN commented on server/index.js:
```diff
@@ -45,8 +45,12 @@ app.get('/:listingID/averageReviewsRating', (req, res) => {
     ratingsValue.shift();
-    ratingsValue.forEach((rating) => rating.toFixed(1))
-    let averageRating = (ratingsValue.reduce((a,b) => a + b, 0 ))/(ratingsValue.length)
+    for(var key in ratings[0]) {
+      if(key !== '_id') {
+        ratings[0][key] = ratings[0][key].toFixed(1)
+      }
+    }
+    let averageRating = (ratingsValue.reduce((a,b) => a + b, 0 ))/(ratingsValue.length);
```
hey @nelsonly123 feel like it would be redundant here to change all the average reviews to a string to add a decimal point only to convert them back to a number and average it all up. i think it would be easier for you to change the seeding script to included the 'precision' option in faker for numbers 

from your seeding script:

````
reviewRating: {
          cleanliness: faker.random.number({min:1, max:5}),
          communication: faker.random.number({min:1, max:5}),
          checkIn: faker.random.number({min:1, max:5}),
          accuracy: faker.random.number({min:1, max:5}),
          location: faker.random.number({min:1, max:5}),
          value: faker.random.number({min:1, max:5}),
        },
````

i believe you can do 
``` 
reviewRating: {
          cleanliness: faker.random.number({min:1, max:5, precision: 0.1}),
          communication: faker.random.number({min:1, max:5, precision: 0.1}}),
          checkIn: faker.random.number({min:1, max:5, precision: 0.1}}),
          accuracy: faker.random.number({min:1, max:5, precision: 0.1}}),
          location: faker.random.number({min:1, max:5, precision: 0.1}}),
          value: faker.random.number({min:1, max:5, precision: 0.1}}),
        },
```

I think this would be the easier approach and we can solve your issue of precision / decimal points prior to even coming into your server route. Thoughts?

nelsonly123 commented:
Hey Alex! Thanks for looking into this. I went ahead and tried the precision option like you suggested, but I still ran into the same issue where the aggregated ratings for each listing will still have pretty long decimals when calculated, such as this the example below. I actually went ahead and updated my getAverageReviewRating to trunc the extra decimals off before sending to the server and removed the for in loop on the server side. 

{
    "averageRating": 3.19,
    "ratings": [
        {
            "_id": "68",
            "avg_clean": 3.3666666666666667,
            "avg_communication": 2.6166666666666667,
            "avg_checkIn": 2.983333333333333,
            "avg_accuracy": 3.016666666666667,
            "avg_location": 3.483333333333333,
            "avg_value": 3.6999999999999997
        }
    ]
}

GalexyN commented:
Thanks for the example @nelsonly123  good to know that the long decimals were still there. Looks like you've found a solution though! Worst case you could probably edit the average rating after you reduced it with toPrecision to cutoff all the extra decimals
maybe something like this

```
let averageRating = ~~(((ratingsValue.reduce((a,b) => a + b, 0 ))/(ratingsValue.length)).toPrecision(2));
```
